packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -82,45 +81,68 @@ function readPackageSync(_: unknown, file: Config.Path): PkgJson {
 }
 
 function createPackageFilter(
-  conditions?: Array<string>,
+  originalPath: Config.Path,
   userFilter?: ResolverOptions['packageFilter'],
 ): ResolverOptions['packageFilter'] {
-  function attemptExportsFallback(pkg: PkgJson) {
-    const options: ResolveExportsOptions = conditions
-      ? {conditions, unsafe: true}
-      : // no conditions were passed - let's assume this is Jest internal and it should be `require`
-        {browser: false, require: true};
-
-    try {
-      return resolveExports(pkg, '.', options);
-    } catch {
-      return undefined;
-    }
+  if (shouldIgnoreRequestForExports(originalPath)) {
+    return userFilter;
   }
 
-  return function packageFilter(pkg, packageDir) {
+  return function packageFilter(pkg, ...rest) {
     let filteredPkg = pkg;
 
     if (userFilter) {
-      filteredPkg = userFilter(filteredPkg, packageDir);
-    }
-
-    if (filteredPkg.main != null) {
-      return filteredPkg;
+      filteredPkg = userFilter(filteredPkg, ...rest);
     }
 
-    const indexInRoot = resolve(packageDir, './index.js');
-
-    // if the module contains an `index.js` file in root, `resolve` will request
-    // that if there is no `main`. Since we don't wanna break that, add this
-    // check
-    if (isFile(indexInRoot)) {
+    if (filteredPkg.exports == null) {
```

Comment:
@SimenB, is it possible to release just this part without waiting for https://github.com/browserify/resolve/issues/253?

Disabling `main` fields of external packages that have [dual/hybrid export](https://2ality.com/2019/10/hybrid-npm-packages.html) is sooo annoying.

Reply:
That would be a breaking change, unfortunately. So not until Jest 28